### PR TITLE
Endpoint to get dependent services and resource groups

### PR DESCRIPTION
**Pull request by OferLevi85**
Signed-off-by: Ofer Levi <ofer.levi@verizonmedia.com>

### CODE REVIEW

File: servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java [hunk 1]
```diff
@@ -10388,6 +10388,43 @@ public ServiceIdentityList getDependentServiceList(ResourceContext ctx, String d
...
+            DependentServiceResourceGroup dependentServiceResourceGroup = new DependentServiceResourceGroup();
+            dependentServiceResourceGroup.setService(service);
+            dependentServiceResourceGroup.setDomain(finalDomainName);
+            dependentServiceResourceGroup.setResourceGroups(getResourceGroupsForProviderServiceTenantDomain(service, finalDomainName));
+            return dependentServiceResourceGroup;
+        }).collect(Collectors.toList());
+        return new DependentServiceResourceGroupList().setServiceAndResourceGroups(dependentServiceResourceGroups);
+    }
+
+    private List<String> getResourceGroupsForProviderServiceTenantDomain(String service, String tenantDomain) {
```
Comment (havetisyan) [inline servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java:10418]: let's mark service and tenantDomain arguments as final

**Review by havetisyan — CHANGES_REQUESTED** [review]

File: servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java [hunk 2]
```diff
@@ -10388,6 +10388,43 @@ public ServiceIdentityList getDependentServiceList(ResourceContext ctx, String d
...
+
+        // for consistent handling of all requests, we're going to convert
+        // all incoming object values into lower case (e.g. domain, role,
+        // policy, service, etc name)
+
+        domainName = domainName.toLowerCase();
+        setRequestDomain(ctx, domainName);
+
+        ServiceIdentityList serviceIdentityList = dbService.listServiceDependencies(domainName);
+        String finalDomainName = domainName;
```
[the remark this answers is not in the document]
Comment (OferLevi85) [reply servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java:10407]: Lambda required a "final" parameter. I can't set the original domainName final as we might change it by lowercasing.

[the remark this answers is not in the document]
Comment (havetisyan) [reply servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java:10407]: but finalDomainName isn't marked final?

[the remark this answers is not in the document]
Comment (OferLevi85) [reply servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java:10407]: Yeah, I apparently making a new variable that never changes is enough as it is considered "effectively final". 
I agree that it isn't very clear so I'll add the final keyword.

**Review by havetisyan — APPROVED** [review]

**Pull request merged**